From 21df0a0bd3453bbd0cbabd99bf81d9753e0b85cd Mon Sep 17 00:00:00 2001
From: sys_icgvert <icgvert@intel.com>
Date: Wed, 23 Mar 2016 14:41:32 -0700
Subject: [PATCH 06/10] atomisp2: css2400 and 2401
 irci_ecr-master_20160323_0200 Merge

CSS2.1 for css2400 and 2401_legacy for
irci_ecr-master_20160323_0200 firmware tag

Change-Id: Id2f78e1c6a5cf39afe618d714efd4d1d9aae81d3
Signed-off-by: sys_icgvert <icgvert@intel.com>
Original-Tracked_On:
Reviewed-on: https://icggerrit.ir.intel.com/109762
Build: Dessert M BXT RVP (SC)
Approver: WeiX Meng <weix.meng@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Build: Dessert M BTNS (SC)
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-26488
Signed-off-by: Tao Wei <wei.tao@intel.com>
Reviewed-on: https://android.intel.com:443/487152
---
 .../drivers/media/pci/atomisp2/css/camera/pipe/src/pipe_binarydesc.c  | 3 +--
 .../external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css.h   | 2 +-
 .../camera/drivers/media/pci/atomisp2/css/ia_css_types.h              | 2 +-
 .../camera/drivers/media/pci/atomisp2/css/ia_css_version_data.h       | 2 +-
 .../media/pci/atomisp2/css/runtime/binary/interface/ia_css_binary.h   | 2 +-
 .../camera/drivers/media/pci/atomisp2/css/sh_css_firmware.c           | 4 ++--
 6 files changed, 7 insertions(+), 8 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/camera/pipe/src/pipe_binarydesc.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/camera/pipe/src/pipe_binarydesc.c
index 1815180..5de33f8 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/camera/pipe/src/pipe_binarydesc.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/camera/pipe/src/pipe_binarydesc.c
@@ -763,8 +763,7 @@ void ia_css_pipe_get_post_gdc_binarydesc(
 static unsigned int pre_de_isp27_modes[MAX_NUM_PRE_DE_STAGES] =
 {
 	IA_CSS_BINARY_MODE_PRE_DE,
-	IA_CSS_BINARY_MODE_PRE_DE_STAGE1,
-	IA_CSS_BINARY_MODE_PRE_DE_STAGE2
+	IA_CSS_BINARY_MODE_PRE_DE_STAGE1
 };
 
 void ia_css_pipe_get_pre_de_binarydesc(
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css.h
index e074ebf..078a0db 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css.h
@@ -1,4 +1,4 @@
-/* Release Version: irci_ecr-master_20160315_0113 */
+/* Release Version: irci_ecr-master_20160323_0200 */
 /**
 Support for Intel Camera Imaging ISP subsystem.
 Copyright (c) 2010 - 2015, Intel Corporation.
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_types.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_types.h
index 99f8575..303ecd5 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_types.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_types.h
@@ -1,4 +1,4 @@
-/* Release Version: irci_ecr-master_20160315_0113 */
+/* Release Version: irci_ecr-master_20160323_0200 */
 /**
 Support for Intel Camera Imaging ISP subsystem.
 Copyright (c) 2010 - 2015, Intel Corporation.
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_version_data.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_version_data.h
index 67fb366..7f935c8 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_version_data.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_version_data.h
@@ -22,7 +22,7 @@ more details.
 #define __IA_CSS_VERSION_DATA_H
 
 
-#define CSS_VERSION_STRING "REL:20160315_11.2_0521; API:2.1.22.2; GIT:irci___#64cb3e22e01a43907f4b6bd2675c1a48a9b4ff1a; SDK:; USER:viedifw; "
+#define CSS_VERSION_STRING "REL:20160323_12.3_0824; API:2.1.22.2; GIT:irci___#68ed6f17feea6cf4c24f5d25684588f180ca6bfe; SDK:; USER:viedifw; "
 
 
 #endif
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/runtime/binary/interface/ia_css_binary.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/runtime/binary/interface/ia_css_binary.h
index 54ab384..b4cfd6b 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/runtime/binary/interface/ia_css_binary.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/runtime/binary/interface/ia_css_binary.h
@@ -52,7 +52,7 @@ more details.
 #define MAX_NUM_PRIMARY_STAGES NUM_PRIMARY_HQ27_STAGES
 
 #define NUM_PRE_DE_STAGES       1 /* number of pre-de binaries can be loaded on non-ISP2.7 pipe */
-#define NUM_PRE_DE_ISP27_STAGES 3 /* number of pre-de binaries can be loaded on ISP2.7 pipe */
+#define NUM_PRE_DE_ISP27_STAGES 2 /* number of pre-de binaries can be loaded on ISP2.7 pipe */
 #define MAX_NUM_PRE_DE_STAGES   NUM_PRE_DE_ISP27_STAGES
 
 /* Indicate where binaries can read input from */
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_firmware.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_firmware.c
index eefd77f..44d0e80 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_firmware.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_firmware.c
@@ -47,10 +47,10 @@ struct fw_param {
 /* Warning: same order as SH_CSS_BINARY_ID_* */
 static struct firmware_header *firmware_header;
 
-/* The string STR(irci_ecr-master_20160315_0113) is a place holder
+/* The string STR(irci_ecr-master_20160323_0200) is a place holder
  * which will be replaced with the actual RELEASE_VERSION
  * during package generation. Please do not modify  */
-static const char *release_version = STR(irci_ecr-master_20160315_0113);
+static const char *release_version = STR(irci_ecr-master_20160323_0200);
 
 #define MAX_FW_REL_VER_NAME	300
 static char FW_rel_ver_name[MAX_FW_REL_VER_NAME] = "---";
-- 
1.9.1

